src/hexkit/providers/akafka/provider.py

```diff
@@ -473,3 +479,4 @@ async def run(self, forever: bool = True) -> None:
         else:
             event = await self._consumer.__anext__()
             await self._consume_event(event)
+        await self._consumer.commit()
```

Comment:
This would just commit once the consumer is stopped, right? But we would like to commit after every successfully consumed event. Would modifiying the _consumer_event method be suitable:

```
    async def _consume_event(self, event: ConsumerEvent) -> None:
        """Consume an event by passing it down to the translator via the protocol."""
        event_label = self._get_event_label(event)
        headers = headers_as_dict(event)
        try:
            type_ = get_header_value(header_name="type", headers=headers)
            correlation_id = get_header_value(
                header_name="correlation_id", headers=headers
            )
        except EventHeaderNotFoundError as err:
            logging.warning("Ignored an event: %s. %s", event_label, err.args[0])
            return
        if type_ in self._types_whitelist:
            logging.info('Consuming event of type "%s": %s', type_, event_label)
            try:
                async with set_correlation_id(correlation_id):
                    # blocks until event processing is completed:
                    await self._translator.consume(
                        payload=event.value,
                        type_=type_,
                        topic=event.topic,
                    )
            except Exception:
                logging.error(
                    "A fatal error occurred while processing the event: %s",
                    event_label,
                )
                raise
            else:
                await self._consumer.commit()
        else:
            logging.info("Ignored event of type %s: %s", type_, event_label)```

Reply:
Yep, put that in the wrong place.
All the places that ignore the event also need a commit, else we constantly reconsume events that are not even processed.

Reply:
Yes right